packages/terra-date-time-picker/src/DateTimePicker.jsx

```diff
@@ -0,0 +1,358 @@
+   */
+  onChangeRaw: PropTypes.func,
+  /**
+   * A callback function to let the containing component (e.g. modal) to regain focus.
+   */
+  releaseFocus: PropTypes.func,
+  /**
+   * A callback function to request focus from the containing component (e.g. modal).
+   */
+  requestFocus: PropTypes.func,
+  /**
+   * An ISO 8601 string representation of the initial value to show in the date and time inputs.
+   */
+  value: PropTypes.string,
+};
+
+const defaultProps = {
+  excludeDates: undefined,
+  filterDate: undefined,
+  includeDates: undefined,
+  inputAttributes: undefined,
+  maxDateTime: undefined,
+  minDateTime: undefined,
+  name: undefined,
+  onChange: undefined,
+  onChangeRaw: undefined,
+  releaseFocus: undefined,
+  requestFocus: undefined,
+  value: undefined,
+};
+
+const contextTypes = {
+  /* eslint-disable consistent-return */
+  intl: (context) => {
+    if (context.intl === undefined) {
+      return new Error('Please add locale prop to Base component to load translations');
+    }
+  },
+};
+
+const keyCodes = {
+  ARROWDOWN: 40,
+};
+
+class DateTimePicker extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      dateTime: DateTimeUtil.createSafeDate(props.value),
+      isAmbiguousTime: false,
+      isTimeClarificationOpen: false,
+    };
+
+    // Track the dateValue and timeValue outside of the react state to limit the number of renderings that occur.
+    this.dateValue = '';
+    this.timeValue = '';
+
+    this.handleDateChange = this.handleDateChange.bind(this);
+    this.handleDateChangeRaw = this.handleDateChangeRaw.bind(this);
+    this.handleTimeChange = this.handleTimeChange.bind(this);
+    this.handleOnSelect = this.handleOnSelect.bind(this);
+    this.handleOnDateBlur = this.handleOnDateBlur.bind(this);
+    this.handleOnTimeBlur = this.handleOnTimeBlur.bind(this);
+
+    this.handleDaylightSavingButtonClick = this.handleDaylightSavingButtonClick.bind(this);
+    this.handleStandardTimeButtonClick = this.handleStandardTimeButtonClick.bind(this);
+  }
+
+  componentWillMount() {
+    const dateTime = DateTimeUtil.createSafeDate(this.props.value);
+    const dateFormat = DateUtil.getFormatByLocale(this.context.intl.locale);
+
+    this.dateValue = DateTimeUtil.formatMomentDateTime(dateTime, dateFormat);
+    this.timeValue = DateTimeUtil.hasTime(this.props.value) ? DateTimeUtil.formatISODateTime(this.props.value, 'HH:mm') : '';
+  }
+
+  handleOnSelect(event, selectedDate) {
+    const dateFormat = DateUtil.getFormatByLocale(this.context.intl.locale);
+    this.dateValue = DateTimeUtil.formatISODateTime(selectedDate, dateFormat);
+    const previousDateTime = this.state.dateTime ? this.state.dateTime.clone() : null;
+    const updatedDateTime = DateTimeUtil.syncDateTime(previousDateTime, selectedDate, this.timeValue);
+
+    if (!previousDateTime || previousDateTime.format() !== updatedDateTime.format()) {
+      this.checkAmbiguousTime(updatedDateTime);
+    }
```

Comment:
Would we want to call this.props.OnChange for selections?

Reply:
When selecting a date from the picker, both onChange and onSelect will be invoked, so handleDateChange will take care of calling this.props.onChange.